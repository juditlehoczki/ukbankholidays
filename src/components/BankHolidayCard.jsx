import React from "react";
import { useNavigation } from "@react-navigation/native";
import { HStack, Icon, IconButton, Text, VStack } from "native-base";
import { MaterialIcons } from "@expo/vector-icons";

import { formatDate } from "../utils/utils";

const BankHolidayCard = ({ item }) => {
  const { navigate } = useNavigation();
  const { bunting, title, date, notes } = item;

  return (
    <HStack
      justifyContent="space-between"
      alignItems="center"
      rounded="lg"
      shadow={8}
      p={2}
      m={2}
      bgColor="orange.100"
    >
      <VStack>
        <HStack>
          {bunting && <Text fontSize="lg">ğŸ‡¬ğŸ‡§ </Text>}
          <Text fontSize="lg">{title}</Text>
        </HStack>
        <Text bold italic>
          {formatDate(date)}
        </Text>
        {!!notes && <Text fontSize="lg">ğŸ—’</Text>}
      </VStack>
      <IconButton
        icon={<Icon as={MaterialIcons} name="edit" />}
        borderRadius="full"
        _icon={{
          color: "orange.500",
          size: "sm",
        }}
        onPress={() => navigate("Edit", { bankHoliday: item })}
      />
    </HStack>
  );
};

export default BankHolidayCard;
