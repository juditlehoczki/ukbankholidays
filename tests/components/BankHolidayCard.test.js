import React from "react";
import { fireEvent, render } from "react-native-testing-library";
import { NativeBaseProvider } from "native-base";

import BankHolidayCard from "../../src/components/BankHolidayCard";
import inset from "../inset";

const mockNavigation = jest.fn();

jest.mock("@react-navigation/native", () => ({
  useNavigation: () => ({
    navigate: mockNavigation,
  }),
}));

describe("BankHolidayCard Component", () => {
  let props;
  beforeEach(() => {
    props = {
      item: {
        title: "Test Title",
        date: "2022-01-30",
        notes: "Test Notes",
        bunting: false,
      },
    };
  });

  test("renders correctly", () => {
    const component = render(
      <NativeBaseProvider initialWindowMetrics={inset}>
        <BankHolidayCard {...props} />
      </NativeBaseProvider>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
  test("Edit icon navigates to Edit screen with item", async () => {
    const component = render(
      <NativeBaseProvider initialWindowMetrics={inset}>
        <BankHolidayCard {...props} />
      </NativeBaseProvider>
    );
    const IconButtonNode = await component.getByA11yRole("button");
    fireEvent.press(IconButtonNode);
    expect(mockNavigation).toHaveBeenCalledWith("Edit", {
      bankHoliday: props.item,
    });
  });
});
