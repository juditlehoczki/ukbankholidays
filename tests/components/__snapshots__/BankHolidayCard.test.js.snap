// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BankHolidayCard Component renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffedd5",
        "borderRadius": 8,
        "elevation": 9,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 8,
        "marginLeft": 8,
        "marginRight": 8,
        "marginTop": 8,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.32,
        "shadowRadius": 5.46,
      }
    }
  >
    <View
      dataSet={Object {}}
      style={Object {}}
    >
      <View
        dataSet={Object {}}
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          dataSet={Object {}}
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#1f2937",
              "fontFamily": undefined,
              "fontSize": 18,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 27,
              "textDecorationLine": undefined,
            }
          }
        >
          Test Title
        </Text>
      </View>
      <View
        dataSet={Object {}}
        style={Object {}}
      />
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#1f2937",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontStyle": "italic",
            "fontWeight": "700",
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": undefined,
          }
        }
      >
        30 January 2022
      </Text>
      <View
        dataSet={Object {}}
        style={Object {}}
      />
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#1f2937",
            "fontFamily": undefined,
            "fontSize": 18,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 27,
            "textDecorationLine": undefined,
          }
        }
      >
        ðŸ—’
      </Text>
    </View>
    <View
      dataSet={Object {}}
      style={Object {}}
    />
    <View
      _text={
        Object {
          "color": "lightText",
        }
      }
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      colorScheme="primary"
      dataSet={Object {}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onHoverIn={[Function]}
      onHoverOut={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 9999,
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
    >
      <Text />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
